src/app/happiness-tracker/happiness-tracker.component.ts

```diff
@@ -1,10 +1,143 @@
+  relationship1: new FormControl('', Validators.required),
+  relationship2: new FormControl('', Validators.required),
+  relationship3: new FormControl('', Validators.required),
+  relationship4: new FormControl('', Validators.required),
+  relationship5: new FormControl('', Validators.required)
+ });
 
+ get f(){
+  return this.form.controls;
+ }
+
+ calculateTimeValues(timeCategory: string | null | undefined){
+  var timeValue: number = 0;
+
+  if(timeCategory === "never" || timeCategory === 'stronglyAgree'){
+    timeValue = 4;
+  } else if(timeCategory === "rarely" || timeCategory === 'agree'){
+    timeValue = 3;
+  } else if(timeCategory === 'sometimes' || timeCategory === 'disagree'){
+    timeValue = 2;
+  } else if(timeCategory === 'frequently' || timeCategory ===  'stronglyDisagree'){
+    timeValue = 1;
+  } else {
+    timeValue = 0;
+  };
+
+  return timeValue;
+ }
+
+ calculateRelationshipValues(timeCategory: string | null | undefined){
+  var timeValue: number = 0;
+
+  if(timeCategory === 'stronglyDisagree' || timeCategory === 'never'){
+    timeValue = 1;
+  } else if(timeCategory === 'disagree' || timeCategory === 'rarely'){
+    timeValue = 2;
+  } else if(timeCategory === 'agree' || timeCategory === 'sometimes'){
+    timeValue = 3;
+  } else if(timeCategory === 'stronglyAgree' || timeCategory === 'frequently'){
+    timeValue = 4;
+  } else{
+    timeValue = 0;
+  }
+
+  return timeValue;
+ }
+
+ calculateProductivityScore(prod1: string | null | undefined, prod2: string | null | undefined, prod3: string | null | undefined, prod4: string | null | undefined, prod5: string | null | undefined){
+  var prod1Value: number = 0;
+  var prod2Value: number = 0;
+  var prod3Value: number = 0;
+  var prod4Value: number = 0;
+  var prod5Value: number = 0;
+  var sum: number = 0;
+
+  prod1Value = this.calculateTimeValues(prod1);
+  prod2Value = this.calculateTimeValues(prod2);
+  prod3Value = this.calculateTimeValues(prod3);
+  prod4Value = this.calculateTimeValues(prod4);
+  prod5Value = this.calculateTimeValues(prod5);
+
+  sum = prod1Value + prod2Value + prod3Value + prod4Value + prod5Value;
+
+  return sum;
+ }
+
+ calculateSatisfactionScore(sas1: string | null | undefined, sas2: string | null | undefined, sas3: string | null | undefined, sas4: string |null | undefined, sas5: string | null | undefined, sas6: string | null | undefined){
+  var sas1Value: number = 0;
+  var sas2Value: number = 0;
+  var sas3Value: number = 0;
+  var sas4Value: number = 0;
+  var sas5Value: number = 0;
+  var sas6Value: number = 0;
+  var sum: number = 0;
+
+  sas1Value = this.calculateTimeValues(sas1);
+  sas2Value = this.calculateTimeValues(sas2);
+  sas3Value = this.calculateTimeValues(sas3);
+  sas4Value = this.calculateTimeValues(sas4);
+  sas5Value = this.calculateTimeValues(sas5);
+  sas6Value = this.calculateTimeValues(sas6);
+
+  sum = sas1Value + sas2Value + sas3Value + sas4Value + sas5Value + sas6Value;
+  return sum;
+ }
```

Comment:
I think you have an opportunity to simplify and better structure code like this. I'd like to see you create a quiz service that has a calculate method on it that accepts an array of these `string | null | undefined` types (which if you see all over the place you may want to just create a `NullableString` type that includes those). Then in that method you can iterate the array and calculate/sum the values. Now your component can have a one-line call to get a score. You may even want to move the code from submit into a service.